source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -1508,6 +1508,40 @@ bool SwiftLanguageRuntime::GetDynamicTypeAndAddress_Class(
   if (error.Fail())
     return false;
 
+  // If LLDB succeeded on its own, check whether RemoteAST succeeded.
+  if (class_type_or_name.GetCompilerType().IsValid()) {
+    swift::remoteAST::RemoteASTContext remote_ast(
+                                         *swift_ast_ctx->GetASTContext(),
+                                         GetMemoryReader());
+
+    swift::Type expected_type =
+      GetSwiftType(class_type_or_name.GetCompilerType());
+    swift::remote::RemoteAddress instance_address(in_value.GetPointerValue());
+    auto metadata_address =
+      remote_ast.getHeapMetadataForObject(instance_address);
+    if (!metadata_address) {
+      llvm::errs() << "Remote AST failed to get metadata address for type "
```

Comment:
We don't use llvm::errs.  This information should go into the types log.  Do something like:

      Log *log(GetLogIfAllCategoriesSet(LIBLLDB_LOG_TYPES));
      if (log)
        log->Printf(...);

Reply:
Okay! I'll use that if I'm going to put in any logging that's meant to be merged.